pkg/collector/corechecks/cluster/orchestrator/manifest_buffer.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && orchestrator
+// +build kubeapiserver,orchestrator
+
+package orchestrator
+
+import (
+	"time"
+
+	"go.uber.org/atomic"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ManifestBufferConfig contains information about buffering manifests.
+type ManifestBufferConfig struct {
+	KubeClusterName             string
+	ClusterID                   string
+	MaxPerMessage               int
+	MaxWeightPerMessageBytes    int
+	MsgGroupRef                 *atomic.Int32
+	BufferedManifestEnabled     bool
+	MaxBufferedManifests        int
+	ManifestBufferFlushInterval time.Duration
+}
+
+// ManifestBuffer is a buffer of manifest sent from all collectors
+// It has a slice bufferedManifests used to buffer manifest and stop channel
+// ManifestBuffer is started as a dedicated thread each time CollectorBundle runs a check
+// and gets stopped after the check is done.
+type ManifestBuffer struct {
+	Cfg               *ManifestBufferConfig
+	ManifestChan      chan interface{}
+	bufferedManifests []interface{}
+	stopCh            chan struct{}
+}
+
+// NewManifestBuffer returns a new ManifestBuffer
+func NewManifestBuffer(chk *OrchestratorCheck) *ManifestBuffer {
```

Comment:
wdyt about adding a test that:
- creates a buffer
- starts it
- enqueues n objects
- flushes it 
- checks the state in between each time?

Reply:
Addressed in [d199d9a](https://github.com/DataDog/datadog-agent/pull/13943/commits/d199d9abfe0ac92b5fbaffb4fc4d0f9a68903fd6)